map/bookmark_manager.cpp

```diff
@@ -1141,6 +1142,53 @@ kml::CompilationType BookmarkManager::GetCompilationType(kml::MarkGroupId id) co
   return compilation->second->GetCategoryData().m_type;
 }
 
+void BookmarkManager::SaveTrackRecording(std::string const & name)
+{
+  CHECK_THREAD_CHECKER(m_threadChecker, ());
+  auto const & tracker = GpsTracker::Instance();
+  CHECK(tracker.IsRecentTrackCollectionInitialized(), ("Track should be initialized"));
+  std::vector<location::GpsTrackInfo> recentTrack = tracker.GetRecentTrackCollection();
+  CHECK(!recentTrack.empty(), ("Recorded track should not be empty"));
+
+  kml::TrackData trackData;
+  std::string trackName = name.empty() ? GenerateTrackRecordingName() : name;
+  kml::SetDefaultStr(trackData.m_name, trackName);
+
+  struct kml::MultiGeometry m_geometry;
+  m_geometry.FromGpsInfoPoints(recentTrack);
```

Comment:
nit: instead of Get... and then From... which create a temporary array of points, an approach like this can be faster and consume less memory/battery:
```
MultiGeometry::LineT line;
tracker.GetRecentTrackCollection([&line](location::GpsTrackInfo const & pt, size_t id)->bool
  {
    line.emplace_back(mercator::FromLatLon(pt.m_latitude, pt.m_longitude));
    return true;
  });
multigeometry.m_lines.push_back(std::move(line));
```